lib/puppet/type/ntp_auth_key.rb

```diff
@@ -0,0 +1,45 @@
+# encoding: utf-8
+
+Puppet::Type.newtype(:ntp_auth_key) do
+  @doc = 'NTP Authentication keys'
+
+  apply_to_all
+  ensurable
+
+  newparam(:name, namevar: true) do
+    desc 'Authentication key ID'
+
+    validate do |value|
+      if ((value.is_a? String) || (value.is_a? Integer)) and value.to_i.between?(1, 65535) then super(value)
+      else fail "value #{value.inspect} is invalid, must be an Integer 1-65535."
```

Comment:
Also, technically titles can't be integers:
```
gestalt% be puppet apply -e 'notify { 1: }'
Error: Evaluation Error: Illegal title type at index 0. Expected String, got Integer
```

It might be worth never asking for an integer, since the only time `name` would be an integer is if someone passes `name => 1` as an attribute instead of as a title.